clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -941,6 +1016,13 @@ static Expected<StringRef> runLLVMToSPIRVTranslation(StringRef File,
   return *TempFileOrErr;
 }
 
+static Expected<StringRef> runLLVMToSPIRVTranslation(StringRef File,
```

Comment:
Just a curiousity question. Why was this modification required?
Seems like functionally equivalent to the old flow and the way it's invoked for thinLTO also seems similar.

Thanks

Reply:
It was because I was hitting multithreading issues, we can't use `StringRefs` or `Args` as we want on any thread, so I computed the args before the thinLTO threads are created to work around the issue. Definitely a hack and will require a real solution at some point.

Reply:
That definitely sounds like a good idea to me. @asudarsa, you mentioned we were seeing 95% pass rate on E2E tests with optimization enabled with the SPIR-V backend. Do you agree with dropping support for llvm-spirv in thinLTO and only supporting the backend?

Reply:
+1 on that. I am planning to call for a quick meeting before we do this. I am hoping to use SPIR-V backend for new offload model while we continue to use SPIR-V translator in the old offloading model. This will make the switch to NOT impact our customers right away.